src/Datadog.Trace/Tags.cs

```diff
@@ -7,28 +7,6 @@ namespace Datadog.Trace
     /// </summary>
     public static class Tags
     {
-        /// <summary>
-        /// The environment of the profiled service.
-        /// </summary>
-        public const string Env = "env";
-
-        /// <summary>
-        /// The name of the integration that generated the span.
-        /// Use OpenTracing tag "component"
-        /// </summary>
-        public const string InstrumentationName = "component";
-
-        /// <summary>
-        /// The name of the method that was instrumented to generate the span.
-        /// </summary>
-        public const string InstrumentedMethod = "instrumented.method";
-
-        /// <summary>
-        /// The kind of span (e.g. client, server). Not to be confused with <see cref="Span.Type"/>.
-        /// </summary>
-        /// <seealso cref="SpanKinds"/>
-        public const string SpanKind = "span.kind";
```

Comment:
I'm not against having proxies in here that point to CoreTags in order to reduce the breaking changes, but I don't really *like* it.

Reply:
Agreed. Why not move the entire `Tags` class to the new assembly?

Reply:
Because we add to it based on new integrations. This seemed like the best way to not pollute core with non-core things.

Reply:
These are ultimately just strings, so I don't think they need to be in the Core assembly. The main concern with that assembly should be types that have a well-formed shape

Reply:
@zacharycmontoya A lot of TraceContext and SpanContext were pretty tangled with many tags but not others.

Reply:
With fresh eyes, I'm pretty happy with the way the changes look which includes several proxies for key things and leaves the rest in the `Datadog.Trace` assembly which can be added as we add integrations